#!/usr/bin/env python
PACKAGE = "waypoint_global_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_vel_x", double_t, 0, "Maximum velocity (m/s)", 0.4, 0.1, 1.5)
gen.add("max_vel_theta", double_t, 0, "Maximum rotational velocity (rad/s)", 0.3, 0.1, 1.0)
gen.add("acc_lim_x", double_t, 0, "Acceleration limit (m/s²)", 0.2, 0.1, 1.0)
gen.add("acc_lim_theta", double_t, 0, "Angular acceleration limit (rad/s²)", 0.3, 0.1, 1.0)
gen.add("inflation_radius", double_t, 0, "Inflation radius for obstacles (m)", 0.3, 0.1, 1.0)
gen.add("obstacle_weight", double_t, 0, "Weight for obstacle cost", 10.0, 1.0, 100.0)
gen.add("resolution", double_t, 0, "Planning resolution (m)", 0.05, 0.01, 0.1)
gen.add("waypoint_tolerance", double_t, 0, "Waypoint tolerance distance (m)", 0.2, 0.05, 1.0)
gen.add("lookahead_distance", double_t, 0, "Lookahead distance for path following (m)", 1.0, 0.5, 3.0)

exit(gen.generate(PACKAGE, "waypoint_global_planner", "WaypointPlanner"))
